frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts

```diff
@@ -60,40 +83,138 @@ export const filtersToQueryNode = (filters: Partial<FilterType>): InsightQueryNo
-        }),
-        interval: filters.interval,
-        series,
-    })
+        })
+    }
 
-    if (isLifecycleFilter(filters) && isLifecycleQuery(query)) {
-        query.lifecycleFilter = objectClean({
+    // trends filter
+    if (isTrendsFilter(filters) && isTrendsQuery(query)) {
+        query.trendsFilter = objectCleanWithEmpty({
+            smoothing_intervals: filters.smoothing_intervals,
+            show_legend: filters.show_legend,
+            hidden_legend_keys: filters.hidden_legend_keys,
+            compare: filters.compare,
+            aggregation_axis_format: filters.aggregation_axis_format,
+            aggregation_axis_prefix: filters.aggregation_axis_prefix,
+            aggregation_axis_postfix: filters.aggregation_axis_postfix,
+            breakdown_histogram_bin_count: filters.breakdown_histogram_bin_count,
+            formula: filters.formula,
             shown_as: filters.shown_as,
+            display: filters.display,
         })
     }
 
+    // funnels filter
+    if (isFunnelsFilter(filters) && isFunnelsQuery(query)) {
+        query.funnelsFilter = objectCleanWithEmpty({
+            funnel_viz_type: filters.funnel_viz_type,
+            funnel_from_step: filters.funnel_from_step,
+            funnel_to_step: filters.funnel_to_step,
+            funnel_step_reference: filters.funnel_step_reference,
+            funnel_step_breakdown: filters.funnel_step_breakdown,
+            breakdown_attribution_type: filters.breakdown_attribution_type,
+            breakdown_attribution_value: filters.breakdown_attribution_value,
+            bin_count: filters.bin_count,
+            funnel_window_interval_unit: filters.funnel_window_interval_unit,
+            funnel_window_interval: filters.funnel_window_interval,
+            funnel_order_type: filters.funnel_order_type,
+            exclusions: filters.exclusions,
+            funnel_correlation_person_entity: filters.funnel_correlation_person_entity,
+            funnel_correlation_person_converted: filters.funnel_correlation_person_converted,
+            funnel_custom_steps: filters.funnel_custom_steps,
+            funnel_advanced: filters.funnel_advanced,
+            layout: filters.layout,
+            funnel_step: filters.funnel_step,
+            entrance_period_start: filters.entrance_period_start,
+            drop_off: filters.drop_off,
+            hidden_legend_keys: filters.hidden_legend_keys,
+        })
+    }
+
+    // retention filter
+    if (isRetentionFilter(filters) && isRetentionQuery(query)) {
+        query.retentionFilter = objectCleanWithEmpty({
+            retention_type: filters.retention_type,
+            retention_reference: filters.retention_reference,
+            total_intervals: filters.total_intervals,
+            returning_entity: filters.returning_entity,
+            target_entity: filters.target_entity,
+            period: filters.period,
+        })
+        // TODO: query.aggregation_group_type_index
```

Comment:
?

Reply:
quick search in the code makes me think it's a top level thing so I think you're right 👍